library/Faktory/Worker.hs

```diff
@@ -61,35 +61,37 @@ runWorker settings f = do
   client <- newClient settings $ Just workerId
   beatThreadId <- forkIOWithThrowToParent $ forever $ heartBeat client workerId
 
-  forever (processorLoop client (settingsQueue settings) f)
+  forever (processorLoop client settings f)
     `catch` (\(_ex :: WorkerHalt) -> pure ())
     `finally` (killThread beatThreadId >> closeClient client)
 
-processorLoop :: FromJSON arg => Client -> Queue -> (arg -> IO ()) -> IO ()
-processorLoop client queue f = do
+processorLoop :: FromJSON arg => Client -> Settings -> (arg -> IO ()) -> IO ()
+processorLoop client settings f = do
   let
     processAndAck job = do
       f $ jobArg job
       ackJob client job
 
-  mJob <- fetchJob client queue
+  emJob <- fetchJob client $ settingsQueue settings
 
-  case mJob of
-    Just job ->
+  case emJob of
+    Left err -> settingsLogError settings $ "Invalid Job: " <> err
+    Right Nothing -> threadDelaySeconds 1
```

Comment:
Why 1 second? Is this configurable?

Reply:
This is moved code, not new. We just wanted _some_ delay so the thing doesn't spin wildly heating up your CPU while there are no Jobs to FETCH. I don't think much thought went into the exact value used.

Reply:
Configurable wouldn't be a bad thing

Reply:
https://github.com/frontrowed/faktory_worker_haskell/issues/27